src/frontend/src/lib/components/ui/Link.svelte

```diff
@@ -0,0 +1,46 @@
+<script lang="ts">
+	import { goto } from '$app/navigation';
+	import { createEventDispatcher } from 'svelte';
+
+	export let href: string;
+	export let ariaLabel: string;
+	export let external = false;
+	export let iconVisible = true;
+	export let inline = false;
+	export let color: 'blue' | 'inherit' = 'inherit';
+	export let fullWidth = false;
+
+	const dispatch = createEventDispatcher();
+
+	// Custom click handler to guarantee that it prevents default browser behaviour (full page reload)
```

Comment:
Isn't using `click` event on a HTML `a` element considered as a bad practice?

From MDN [documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a):

> Use a [<button>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button) instead. **In general, you should only use a hyperlink for navigation to a real URL.**

Also isn't this hack having for side effect to push twice the targeted page in the history? once with the href and once with the click?

I think it's cleaner and safer to use link strictly for navigation and not try to hijack browser and web API behavior.

Reply:
The issue is that if we use it for internal navigation, and inside the popover, it will always refresh the page. remember you had said once that it should be `<a>`, but maybe in that case I shall leave a `<button>` element just for the settings menu item?

Reply:
Yeah a link but without click 😅

Reply:
Would using a link and closing the popover on mount in settings be a not to dirty option? Just thinking out loud

Reply:
Or a `beforeNavigate`

Reply:
I am using `beforeNavigate` in the last commit, but the issue persist: it is refreshing the page when I click on it, because inside the popover the links are treated as native behavior, re-routing the page, and refreshing everything.

Shall I leave the `Settings` item as button, instead of link?

Reply:
Unless we have a web standard compleant solution, I would keep the button yes.

Reply:
It was just for `a11y` reasons that you mentioned one. I will close this one